src/app/modules/assignments/components/AssignmentForm.tsx

```diff
@@ -1,32 +1,94 @@
 import * as React from 'react'
 import { Form } from 'react-bootstrap';
-import { Field } from 'redux-form';
-import { Panel } from 'react-bootstrap';
-import { required } from '../../../infrastructure/Validators';
-import { TextFormField, QualificationsChecklist, AssignmentTypeSelector } from '../../../components/Form';
+import { 
+    Field, 
+    InjectedFormProps
+} from 'redux-form';
+import {
+    TextFormField,
+    RequiredTrainingChecklist,
+    DateTimeField,
+    NumberOfSheriffsSelector,
+    CheckboxField,
+    TextArea
+} from '../../../components/Form';
+import { Validators } from '../../../infrastructure';
+
+interface GateSecurityFieldsProps{
+
+}
+class GateSecurityFields extends React.Component<GateSecurityFieldsProps, any>{
+    render() {
+        return (
+            <div>
+                <Field name="gateNumber" component={TextFormField} label="Gate Number" />
+            </div>
+        );
+    }
+}
+
+interface EscortServiceFieldsProps{
+
+}
+class EscortServiceFields extends React.Component<EscortServiceFieldsProps, any>{
+    render() {
+        return (
+            <div>
+                <Field name="pickupLocation" component={TextFormField} label="Pick-Up Location" />
+                <Field name="dropoffLocation" component={TextFormField} label="Drop-Off Location" />
+            </div>
+        );
+    }
+}
+interface CourtSecurityFieldsProps{
+
+}
+
+class CourtSecurityFields extends React.Component<CourtSecurityFieldsProps, any>{
+    render() {
+        return (
+            <div>
+                <Field name="courtRoom" component={TextFormField} label="Court Room" />
+                <Field name="assignmentCourt" component={CheckboxField} value="tesing this value" label="Assignment Court" />
+            </div>
+        );
+    }
+}
 
 export interface AssignmentFormProps {
-   handleSubmit:()=>void;
+    handleSubmit?: () => void;
+    onSubmitSuccess?: () => void;
+    showCourtSecurityFields?: boolean;
+    showDocumentSericesFields?: boolean;
+    showEscortServicesFields?: boolean;
+    showGateSecurityFields?: boolean;
```

Comment:
Since these options are mutually exclusive, I think an enum probably be best in this case. Or, we could even use the AssignmentTypes that are in the API.  (This is something that will most likely live in the API at some point.) Anyways, for the sake of the MVP this will do, but we'll want to find a lot of these constants like AssignmentTypes etc into the definition of the API and then consume them from components like these.